### PR TITLE
Better logging for SSH connections and errors

### PR DESCRIPTION
**Pull request by jkeiser**

### CODE REVIEW

File: lib/kitchen/transport/ssh.rb [hunk 1]
```diff
@@ -220,16 +218,39 @@ def wait_until_ready
...
+
           Net::SSH.start(hostname, username, options)
+
         rescue *RESCUE_EXCEPTIONS_ON_ESTABLISH => e
+          # Make a much better explanation for why
+          error_type, explanation = case e
+          when Errno::ENETUNREACH, Errno::EHOSTUNREACH
+            [ "Unable to reach host", "The could mean the server has no public IP or are behind a VPN you aren't connected to." ]
+          when Errno::ETIMEDOUT, Timeout::Error, Net::SSH::ConnectionTimeout
+            [ "Connection timeout", "Could not connect within #{options[:timeout]}s. This could mean SSH hasn't started yet, or it could mean server's network acls don't allow external connections to port #{port}." ]
```
Comment (thommay) [inline lib/kitchen/transport/ssh.rb]: Nit: `#{options[:timeout]} seconds.`

**Comment by thommay** [comment]
tiny thing, otherwise :+1:

**Comment by mwrock** [comment]
:+1:

**Comment by mwrock** [comment]
quite a few travis failures

**Comment by lamont-granquist** [comment]
just think we're not going to have time to ressurect this one.

**Pull request closed without merging**